CourseApp/Cat.cs

```diff
@@ -0,0 +1,75 @@
+﻿namespace CourseApp
+{
+    using System;
+
+    public class Cat
+    {
+        private string name;
+        private double tailLength;
+
+        public Cat(int age, string name, double tailLength)
+        {
+            if (age < 0)
+            {
+                throw new ArgumentException();
+            }
+            else
+            {
+                this.Age = age;
+            }
+
+            if (name == string.Empty || name.Length == 0)
+            {
+                throw new System.ArgumentException();
+            }
+            else
+            {
+                this.name = name;
+            }
+
+            if (tailLength < 0)
+            {
+                throw new System.ArgumentException();
+            }
+            else
+            {
+                this.tailLength = tailLength;
+            }
+        }
+
+        public Cat()
+        {
+            this.Age = 0;
+            this.Name = "Not named :c";
+            this.TailLength = 0;
+        }
+
+        public int Age { get; set; }
+
+        public string Name { get => name; set => name = value; }
+
+        public double TailLength { get => tailLength; set => tailLength = value; }
```

Comment:
Let's move to the start of file 🥺

Reply:
@jskonst ?

Reply:
guess this is required by stylecop default rules (not sure)